libkbfs/mdserver_memory.go

```diff
@@ -336,6 +348,16 @@ func (md *MDServerMemory) Put(ctx context.Context, rmds *RootMetadataSigned,
 		return MDServerError{err}
 	}
 
+	if extra == nil {
```

Comment:
Ah, never mind, `md` is the actual server itself, I see. So the idea is that `extra` is non-nil only when there's a new `extra` object, and otherwise the client sends a `nil` `extra` object and a lookup is done via the IDs. Got it.